pkg/config/mesh/mesh.go

```diff
@@ -231,13 +234,14 @@ func mergeMap(original map[string]string, merger map[string]string) map[string]s
 	if original == nil && merger == nil {
 		return nil
 	}
-	if original == nil {
-		original = map[string]string{}
+	result := map[string]string{}
```

Comment:
the new map` result` is not required

Reply:
this function usages expect `original` param to remain the same. this isn't possible without creating a new map for result